### PR TITLE
[receiver/kafkareceiver] Add otlp_json support in kafka receiver

### PR DESCRIPTION
**Pull request by joeyyy09**
**Description:** <Describe what has changed.>
The current features dont support otlp_json in receivers/kafkareceivers. 
Add support for otlp_json which accepts json formated for Otel Collector kafka receiver
**Link to tracking Issue:** <Issue number if applicable>  #33627

**Testing:** <Describe what testing was performed and which tests were added.>
Added test files for the same. 

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Comment by joeyyy09** [comment]
this is a draft in process, would love to see if im going in the right direction or not. have looked if i can find the required parsing logic in the codebase but couldnt find a lot that was reusable. I've been also looking into how the awss3 receiver has been parsing otlp_json and found one there but not sure if i'm correct to be able to reuse that

File: receiver/kafkareceiver/unmarshaler.go
Comment (yurishkuro) [inline receiver/kafkareceiver/unmarshaler.go:53]: Do you need a new class here or can you reuse the one above and pass it a pdata json unmarshaler?

File: receiver/kafkareceiver/README.md
Comment (yurishkuro) [inline receiver/kafkareceiver/README.md:34]: This is a user facing documentation. It needs to describe the function, not implementation details.

File: receiver/kafkareceiver/unmarshaler.go
Comment (joeyyy09) [reply receiver/kafkareceiver/unmarshaler.go:53]: Can reuse the above one but did for better code clarity, removing that.

File: receiver/kafkareceiver/README.md
Comment (joeyyy09) [reply receiver/kafkareceiver/README.md:34]: Understood

File: receiver/kafkareceiver/unmarshaler.go
Comment (yurishkuro) [reply receiver/kafkareceiver/unmarshaler.go:53]: I don't think adding an extra indirection is improving code quality. The otlpPb object above is instantiated inline, you can do the same for JSON.

File: receiver/kafkareceiver/unmarshaler.go
Comment (yurishkuro) [inline receiver/kafkareceiver/unmarshaler.go:81]: move to line after otlpPb, for consistency

Comment (joeyyy09) [reply receiver/kafkareceiver/unmarshaler.go]: understood

File: receiver/kafkareceiver/unmarshaler.go
Comment (joeyyy09) [reply receiver/kafkareceiver/unmarshaler.go:53]: Umm, saw other encodings( zipkins ) following a similar structure so did that. Can instantiate inline as well. Which one would you prefer me to follow?

Comment (joeyyy09) [reply receiver/kafkareceiver/unmarshaler.go]: done.

**Review by yurishkuro — APPROVED** [review]

**Comment by yurishkuro** [comment]
@joeyyy09 check the linter failures
`kafkareceiver/unmarshaler.go:61: File is not `gci`-ed with --skip-generated -s standard -s default -s prefix(github.com/open-telemetry/opentelemetry-collector-contrib) (gci)`

**Comment by joeyyy09** [comment]
> @joeyyy09 check the linter failures `kafkareceiver/unmarshaler.go:61: File is not `gci`-ed with --skip-generated -s standard -s default -s prefix(github.com/open-telemetry/opentelemetry-collector-contrib) (gci)`

My bad, fixed that.

File: receiver/kafkareceiver/README.md
Comment (fatsheep9146) [inline receiver/kafkareceiver/README.md:34]: small nit: missing period at the end.

**Review by fatsheep9146 — APPROVED** [review]

**Review by yurishkuro — APPROVED** [review]

**Comment by thmshmm** [comment]
Support for encoding extensions in the kafkareceiver was just merged with https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33888. Using the `otlpencodingextension` should now do the job.

**Comment by yurishkuro** [comment]
Prom tests are failing on main, not an issue here (https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/35119). Everything else in the PR is green. @open-telemetry/collector-maintainers  this can be merged.

@thmshmm I think an upgrade to `otlpencodingextension` is an independent issue and can be tackled separately, there's no need for it to block this PR.

**Pull request merged**
